### ISSUE TITLE
Regression 3.4.1 for ESM environments: TS1203 

### ISSUE THREAD

**Issue by ClementValot** [Avansai/properties-file#22]
```
../../../node_modules/properties-file/lib/esm/properties-file.d.ts:7:3 - error TS1203: Export assignment cannot be used when targeting ECMAScript modules. Consider using 'export default' or another module format instead.

7   export = keyValuePairObject
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~


Found 1 error in ../../../node_modules/properties-file/lib/esm/properties-file.d.ts:7
```

`package.json`
```
blah 
"type": "module"
blah
"dependencies": {
  "properties-file": "^3.4.1",
},
"devDependencies": {
  "typescript": "^5.3.3"
}
```

`tsConfig.json`
```
{
  "compilerOptions": {
    "module": "NodeNext",
    "strict": true,
    "target": "ES6",
    "resolveJsonModule": true
  },
  "exclude": ["node_modules"]
}
```

Lmk if you need more elements for debug but at first sight it will probably be an easy one :)

**Comment by nbouvrette** [comment]
@ClementValot Thanks for reporting this; I'll take a look shortly.

I thought after multiple abandoned tries that I had finally found the correct way to provide dual CommonJS/ESM support, but the surprises never end 😅

**Comment by nbouvrette** [comment]
I found out why my test was working... my `tsconfig` has `"skipLibCheck": true` for some reason... not really the best solution so I'll try to find a better one that doesn't require special configuration

**Comment by nbouvrette** [comment]
I don't know how familiar you are with ESM, but if I understood correctly the way they handle default export is... not great.

I think the best solution will be to change the way the Webpack loader works so instead of doing:

```ts
import * as exampleProperties from "./example.properties";
```

We can do 

```ts
import { properties as exampleProperties } from "./example.properties";
```

Basically use a named import instead which is consistent for both CommonJS and ESM.

I'll have to release a breaking change version to fix this, but it shouldn't be too bad

**Comment by nbouvrette** [comment]
Ok well I just released [3.5.0](https://github.com/Avansai/properties-file/releases/tag/3.5.0) and unless I missed something it should fix this issue (and requires changes for those using the Webpack loader).

Thanks for reporting this!

**Comment by ClementValot** [comment]
Confirmed that 3.5.0 fixes this issue with my setup ✅ 
Thank you for the quick fix!
